### PR TITLE
API Cleanup

### PR DESCRIPTION
**Pull request by pakrym**
1. Move WriteAsync to PipeWriter
2. Move PosisionOf to extension class
3. Remove Get<> from Position

### CODE REVIEW

File: src/System.IO.Pipelines/System/IO/Pipelines/PipeWriter.cs [hunk 1]
```diff
@@ -33,5 +36,24 @@ public abstract class PipeWriter: IOutput
         public abstract void Advance(int bytes);
         public abstract Memory<byte> GetMemory(int minimumLength = 0);
         public abstract Span<byte> GetSpan(int minimumLength = 0);
+
+        public virtual Task WriteAsync(ReadOnlyMemory<byte> source)
```
Comment (pakrym) [inline src/System.IO.Pipelines/System/IO/Pipelines/PipeWriter.cs]: Not entirely sure about this, maybe it needs to return `ValueTask<FlushResult>`

File: src/System.Text.Http.Parser/HttpParser.cs [hunk 2]
```diff
@@ -323,7 +323,7 @@ public HttpParser(bool showErrorDetails)
                                 var current = reader.Position;
                                 var subBuffer = buffer.Slice(reader.Position);
                                 // Split buffers
-                                var lineEnd = subBuffer.PositionOf(ByteLF);
+                                var lineEnd = ReadOnlyBufferExtensions.PositionOf(subBuffer, ByteLF);
```
Comment (davidfowl) [inline src/System.Text.Http.Parser/HttpParser.cs]: ?

File: src/System.IO.Pipelines/System/IO/Pipelines/PipeWriter.cs (hunk 1 above)
Comment (davidfowl) [reply src/System.IO.Pipelines/System/IO/Pipelines/PipeWriter.cs]: Yea, lets do ValueAwaiter<FlushResult>

File: src/System.Text.Http.Parser/HttpParser.cs (hunk 2 above)
Comment (pakrym) [reply src/System.Text.Http.Parser/HttpParser.cs]: It compares two implementations of `PositionOf` on ISequence and ROB, now when both of them are extension methods invocation is ambiguous

File: src/System.IO.Pipelines/System/IO/Pipelines/PipeWriter.cs [hunk 3]
```diff
@@ -11,6 +12,8 @@ namespace System.IO.Pipelines
     /// </summary>
     public abstract class PipeWriter: IOutput
     {
+        private static readonly Task _completedTask = Task.FromResult(0);
```
Comment (davidfowl) [inline src/System.IO.Pipelines/System/IO/Pipelines/PipeWriter.cs]: Remove this.

File: src/System.IO.Pipelines/System/IO/Pipelines/PipeWriter.cs (hunk 1 above)
Comment (pakrym) [reply src/System.IO.Pipelines/System/IO/Pipelines/PipeWriter.cs]: Fixed

File: src/System.Buffers.Primitives/System/Buffers/ReadOnlyBufferExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Buffers
+{
+    public static class ReadOnlyBufferExtensions
```
Comment (KrzysztofCwalina) [inline src/System.Buffers.Primitives/System/Buffers/ReadOnlyBufferExtensions.cs]: If we want to have this as an extension member, I would prefer to have it on MemoryExtensions class (an existing extensions method holder). I don't think we should be adding several extension method holders.

Comment (pakrym) [reply src/System.Buffers.Primitives/System/Buffers/ReadOnlyBufferExtensions.cs]: We can move it to that type during corefx move.

Comment (KrzysztofCwalina) [reply src/System.Buffers.Primitives/System/Buffers/ReadOnlyBufferExtensions.cs]: ok

File: src/System.Text.Http.Parser/HttpParser.cs (hunk 2 above)
Comment (davidfowl) [reply src/System.Text.Http.Parser/HttpParser.cs]: Isn't this a problem?

Comment (pakrym) [reply src/System.Text.Http.Parser/HttpParser.cs]: No, the other extension method doesn't ship it's in buffers.Experimental

File: src/System.Buffers.Primitives/System/Buffers/ReadOnlyBufferExtensions.cs (hunk 4 above)
Comment (pakrym) [reply src/System.Buffers.Primitives/System/Buffers/ReadOnlyBufferExtensions.cs]: They are in different namespaces. MemoryExtensions is in System.

**Pull request merged**